### PR TITLE
Compile final lib to ES5 syntax

### PR DESCRIPTION
**Pull request by eschaefer**
This PR serves an ES5 version of the lib from `dist/index.js`. Should make life easier for webpack/UglifyJS users who typically omit their `node_modules` from transpiling ES6 -> ES5.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015", "stage-0"]
```
Comment (ahdinosaur) [inline .babelrc:2]: this module uses [`es2040`](https://github.com/ahdinosaur/es2040) as specified in `browserify.transform` in `package.json` (this works for my purposes since browserify transpiles while bundling based on this field). we don't need to include the kitchen sink only use preset https://github.com/ahdinosaur/babel-preset-es2040.

File: dist/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+'use strict';
```
Comment (ahdinosaur) [inline dist/index.js]: hm... on the one hand i don't think we should commit this file. on the other hand it's necessary when referencing this module as a git dependency, so we probably should. until now i've completed avoided comitting any transpiled code in my modules, instead using `browserify` as it's intended.

File: index.js [hunk 3]
```diff
@@ -12,9 +12,7 @@ const defaultPassThrough = [
   'data'
 ]
 
-module.exports = HyperFela
-
-function HyperFela ({ h, renderRule }) {
+module.exports = function HyperFela ({ h, renderRule }) {
```
Comment (ahdinosaur) [inline index.js]: hey, what's the deal with these style changes? if it's to conform to `standard` i'm okay with them, but these seem unrelated and i prefer the existing style.

File: package.json [hunk 4]
```diff
@@ -29,21 +28,34 @@
   },
   "homepage": "https://github.com/ahdinosaur/hyper-fela#readme",
   "devDependencies": {
+    "babel-cli": "^6.24.1",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-stage-0": "^6.24.1",
     "budo": "^9.4.7",
     "dependency-check": "^2.7.0",
+    "eslint": "^4.1.1",
```
Comment (ahdinosaur) [inline package.json]: why add `eslint` to `devDependencies`? we already have `standard` as the `test:lint` script.

File: test/index.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 const test = require('tape')
 
-const hyperFela = require('../')
+const hyperFela = require('../index.js')
```
Comment (ahdinosaur) [inline test/index.js]: why?

File: .eslintrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (ahdinosaur) [inline .eslintrc:1]: why add this file? we're already using `standard` in our `test:lint` script. is this to support something in your development environment?

**Review by ahdinosaur — COMMENTED** [review]
hey @eschaefer, thanks for the pull request!

to be honest, i'm not at all excited with this change, since i'm happy using `browserify` ([webpack for browserify users](https://gist.github.com/substack/68f8d502be42d5cd4942)) which supports transpiling this module during the bundling process via the `browserify.transform` field in `package.json`. so this will have negative effects on my own developer experience (bye bye original source maps), plus add git repository bloat and an extra step to forget when publishing.

but... i appreciate you putting in the energy to add this support, since there's no denying tools like `webpack` are popular, and there's no point in me being "right" when trying to successfully collaborate with others.

gave some comments on details within the change. happy for any feedback in response.

cheers!

File: package.json [hunk 7]
```diff
@@ -1,21 +1,20 @@
...
   "scripts": {
     "start": "budo example:bundle.js -d example --live -- -d",
     "test:deps": "dependency-check . && dependency-check . --extra --no-dev -i es2040",
     "test:lint": "standard",
     "test:node": "NODE_ENV=test run-default tape test/*.js --",
     "test:coverage": "NODE_ENV=test nyc npm run test:node",
     "test:coverage:report": "nyc report --reporter=lcov npm run test:node",
-    "test": "npm-run-all -s test:node test:lint test:deps"
+    "test": "npm-run-all -s test:node test:lint test:deps",
+    "build": "babel index.js --out-file dist/index.js"
```
Comment (ahdinosaur) [inline package.json]: we should add a `prepublishOnly` script which runs `npm run build`, otherwise i'm afraid i'd forget to run this step since i don't do this on any of my other modules.

File: index.js (hunk 3 above)
Comment (ahdinosaur) [reply index.js]: by the way, some cultural context on conflicting style changes in a pull request: https://groups.google.com/forum/#!msg/nodejs/MWaivVTirPY/0pnRjKsggkIJ

File: package.json (hunk 4 above)
Comment (eschaefer) [reply package.json]: I use Atom as an editor, and this resolved some in-editor linting plugin errors since I don't have `eslint` installed globally. Doesn't need to be included I guess.

File: index.js (hunk 3 above)
Comment (eschaefer) [reply index.js]: No reason, just had the habit of not referencing functions that aren't explicitly declared beforehand. Can revert.

File: dist/index.js (hunk 2 above)
Comment (eschaefer) [reply dist/index.js]: You are right! I just found out about `prepublish` task that should generate the transpiled version on installation.

File: .babelrc (hunk 1 above)
Comment (eschaefer) [reply .babelrc:2]: I just went with the `babel-cli` to do this since it's more community-supported and doesn't require `browserify`, which only some people use, and isn't defined in the dependencies.

File: test/index.js (hunk 5 above)
Comment (eschaefer) [reply test/index.js]: I had an error when trying to run the tests without it, but this seems to have fixed itself. Reverting. 🙃

File: .eslintrc (hunk 6 above)
Comment (eschaefer) [reply .eslintrc:1]: Yep, added for Atom/Sublime/etc to pick up on the linting config. Not required for everyone I suppose.

**Comment by eschaefer** [comment]
Hey @ahdinosaur thanks for the feedback. 

> i'm happy using browserify (webpack for browserify users) which supports transpiling this module during the bundling process via the browserify.transform field in package.json. so this will have negative effects on my own developer experience (bye bye original source maps), plus add git repository bloat and an extra step to forget when publishing.

I think it makes sense to provide an `npm` package which doesn't care which bundler you use, if any. 

`babel-cli` solves this nicely in the `prepublish` task. Then this _just works_. Seem pretty arbitrary to only want to support users who have `browserify` installed globally. 

Not sure how this affects source maps though. Most `npm` dependencies I've seen are mangled/minified/transpiled. 

In the end, I just wanted this to work in my project, which is 100% as generic as they come. Any lever I can push to increase compatibility seemed worthwhile ✌️

File: .babelrc (hunk 1 above)
Comment (ahdinosaur) [reply .babelrc:2]: hey, i'm happy with `babel-cli`, i'm suggesting we use https://github.com/ahdinosaur/babel-preset-es2040 instead of `babel-preset-es2015` and `babel-preset-stage-0`.

File: .eslintrc (hunk 6 above)
Comment (ahdinosaur) [reply .eslintrc:1]: is it okay if we remove it then?

**Comment by ahdinosaur** [comment]
hey @eschaefer, ended up with an alternative pull request using my preferred approach: https://github.com/ahdinosaur/hyper-fela/pull/2. i appreciate your help and energy on this!

**Comment by ahdinosaur** [comment]
published an es5 compatible version of `hyper-fela` (from #2) as `1.5.0` :tada: 

thanks again for you help @eschaefer, appreciate if you can let me know whether `1.5.0` works for you.

File: .babelrc (hunk 1 above)
Comment (eschaefer) [reply .babelrc:2]: Just curious what the benefit is of supporting just a handpicked subset of ES6? If you're using a powerful transpiler anyway, what do contributors gain from only using a subset?

Comment (ahdinosaur) [reply .babelrc:2]: a technical take: everything is a trade-off: more transpiler features means more context to understand, more surface area to maintain, and slower to run. i'd rather chose what i want to depend on, and encourage others to do the same. said another way, what's the point of being dependent on the entire kitchen sink when all you need is a few specific tools?

a more personal take: i write open source modules as a way to share my personal energy and creativity, to play with [mad science](https://github.com/ahdinosaur/mad-science-handbook). i get bad feelings from many of the new "modern JavaScript" features which i find to be too complex, hard to debug, and have unintended edge cases. so i want this module to feel good for me as the primary maintainer (this is the first pull request here :sparkling_heart:) and to encourage contributors to follow the best development practices that i'm aware of.
